libs/application-generic/src/utils/generate-id.ts

```diff
@@ -0,0 +1,8 @@
+import { customAlphabet } from 'nanoid';
+
+export const ALPHABET = '0123456789abcdefghijklmnopqrstuvwxyz';
```

Comment:
The alphabet is up for debate. I created this one as this is what I understood the requirements to be. The probability of betting the same workflow ID is 1 to 1,679,616. I created a graceful way to retry twice to create a unique ID. If needed, we can increase the number of retries.